packages/language-php/grammars/tree-sitter/queries/highlights.scm

```diff
@@ -233,19 +233,56 @@
 
 (class_declaration (name) @entity.name.type.class.php)
 
+(class_declaration (base_clause (name) @entity.other.inherited-class.php))
+(class_declaration (class_interface_clause (name) @entity.other.implemented-interface.php))
+
+; usage or consumption of traits
+(use_declaration (name) @entity.name.type.trait.php)
+; in use lists, "bar" in `Foo::bar` is a method, not a constant
+(use_list
+  (_
+    (class_constant_access_expression (name) @support.other.function.method.php .)
+    (name) @support.other.function.method.php
+  )
+  (#set! capture.final true)
+)
```

Comment:
Docs for trait usage: https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.visibility

Reply:
Here's what I've come up with:

```scm
; The "Foo" and "Bar" in `use Bar, Foo;`
(use_declaration (name) @support.other.trait.php)

; The "Foo" in `Baz::thud insteadof Foo;`
(use_instead_of_clause (name) @support.other.trait.php .
  (#set! capture.final true))

; In `use` lists, the "bar" in `Foo::bar` is a method, not a constant.
(use_list
  (_
    (class_constant_access_expression (name) @support.other.function.method.php .)
  )
  (#set! capture.final true))
```

* Traits should be `entity` when declared and `support` when referenced
* The right side of an `insteadof` operator seems to be a trait, not a method, so I dropped the content of your line 254; let me know if I'm missing something here

Reply:
> right side of an insteadof operator seems to be a trait, not a method

Oooh, right. Good catch.

> `entity` when declared and `support` when referenced

I promise I'm *trying* to remember all of this. 😄

Reply:
It helps to be old (like me) and to have been a TextMate user and bundle developer.

In TextMate, the `entity` namespace was what populated the symbols list (with some exceptions). If something was scoped with `entity`, you could choose it from a menu and jump to that line. That's what engraved it into my brain.